tests/spread/basic/task.yaml

```diff
@@ -0,0 +1,36 @@
+summary: Ensure that things worked
+
+execute: |
+  # for various utilities
+  . "$TESTSLIB/nested.sh"
+
+  # Start the nested UC20 VM
+  start_nested_core_vm_unit
+
+  # At this point we are able to SSH to the nested VM, so things probably worked
+  # but we should wait for snapd to finish seeding anyways, but first we need to
+  # wait for the snap command to exist
+  retry=400
+  wait=1
+  while ! execute_remote command -v snap; do
+      retry=$(( retry - 1 ))
+      if [ $retry -le 0 ]; then
+          echo "Timed out waiting for snap command to be available. Aborting!"
+          exit 1
+      fi
+      sleep "$wait"
+  done
+
+  # Now formally wait for seeding
+  execute_remote sudo snap wait system seed.loaded
+
+  # We should be in run mode
+  execute_remote cat /proc/cmdline | MATCH snapd_recovery_mode=run
+
+  # We should have a modeenv
+  execute_remote test -f /var/lib/snapd/modeenv
+
+  # TODO: what else qualifies a "successful boot"? 
+  # * no failed systemd units?
+  # * all expected partitions mounted?
+  # * systemd not in degraded mode?
```

Comment:
LUKS has been setup for ubuntu-data, when expected.

TPM measurements were completed.

snapd has sealed things.

Reply:
Sure, I will leave it to you to write such tests in followups, I just left this as a clear entry-point for foundations to take over testing and write whatever tests are appropriate